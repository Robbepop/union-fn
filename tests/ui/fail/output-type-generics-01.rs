fn main() {}

#[union_fn::union_fn]
trait Foo {
    type Output<T> = T;
    
    fn foo() -> Self::Output {}

    type Output = i64;
}
