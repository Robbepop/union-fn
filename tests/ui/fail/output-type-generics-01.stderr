error: cannot have generics for Output type in #[union_fn] trait
 --> tests/ui/fail/output-type-generics-01.rs:5:16
  |
5 |     type Output<T> = T;
  |                ^^^
